sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -210,55 +236,110 @@ class AnalyzeHealthcareEntitiesResultItem(DictMixin):
+        "is_error={})".format(
             self.id,
             self.entities,
+            self.relations,
             self.warnings,
-            self.statistics,
+            repr(self.statistics),
             self.is_error
         )[:1024]
 
+class HealthcareRelation(DictMixin):
+    """HealthcareRelation is a result object which represents a relation detected in a document.
+
+    Every HealthcareRelation is an entity graph of a certain relation type,
+    where all entities are connected and have specific roles within the relation context.
+
+    :ivar relation_type: The type of relation, i.e. the relationship between "100mg" and
+        "ibuprofen" in the document "The subject took 100 mg of ibuprofen" is "DosageOfMedication".
+    :vartype relation_type: str or ~azure.ai.textanalytics.HealthcareEntityRelationType
+    :ivar roles: The roles present in this relation. I.e., in the document
+        "The subject took 100 mg of ibuprofen", the present roles are "Dosage" and "Medication".
+    :vartype roles: list[~azure.ai.textanalytics.HealthcareRelationRole]
+    """
+
+    def __init__(self, **kwargs):
+        self.relation_type = kwargs.get("relation_type")
+        self.roles = kwargs.get("roles")
+
+    @classmethod
+    def _from_generated(cls, healthcare_relation_result, entities):
+        roles = [
+            HealthcareRelationRole._from_generated(r, entities)  # pylint: disable=protected-access
+            for r in healthcare_relation_result.entities
+        ]
+        return cls(
+            relation_type=healthcare_relation_result.relation_type,
+            roles=roles,
+        )
+
+    def __repr__(self):
+        return "HealthcareRelation(relation_type={}, roles={})".format(
+            self.relation_type,
+            repr(self.roles),
+        )[:1024]
+
+class HealthcareRelationRole(DictMixin):
+    """A model representing a role in a relation.
+
+    For example, in "The subject took 100 mg of ibuprofen",
+    "100 mg" is a dosage entity fulfilling the role "Dosage"
+    in the extracted relation "DosageofMedication".
+
+    :ivar name: The role of the entity in the relationship. I.e., in the relation
+        "The subject took 100 mg of ibuprofen", the dosage entity "100 mg" has role
+        "Dosage".
+    :vartype name: str or ~azure.ai.textanalytics.HealthcareEntityRelationRoleType
+    :ivar entity: The entity that is present in the relationship. For example, in
+        "The subject took 100 mg of ibuprofen", this property holds the dosage entity
+        of "100 mg".
+    :vartype entity: ~azure.ai.textanalytics.HealthcareEntity
+    """
+
+    def __init__(self, **kwargs):
+        self.name = kwargs.get("name")
+        self.entity = kwargs.get("entity")
+
+    @staticmethod
+    def _get_entity(healthcare_role_result, entities):
+        nums = _get_indices(healthcare_role_result.ref)
+        entity_index = nums[1]
```

Comment:
do u think a comment with an example of what is returned and why using the index `1` value is needed to help clarify?

Reply:
sure I can add a comment here